pipeline/transformer/transformer_sierra/src/main/scala/weco/pipeline/transformer/sierra/transformers/OntologyTypeOps.scala

```diff
@@ -0,0 +1,212 @@
+      // sourceIdentifier of their own.
+      original.id.allSourceIdentifiers.headOption
+
+    /**
+      * Return a copy of the Subject, with its ontologyType changed to newType.
+      *
+      * In a Subject, the ontologyType needs to be changed in up to two places.
+      * The subject's own sourceIdentifier, and, only in the case simple subjects
+      * the sole Concept in the Subject's concepts list
+      *
+      * In a compound subject, the Concepts in the concepts list will all be label-derived
+      * and do not have an authoritative identifier to match against, so cannot be unified.
+      */
+    def copyWithNewType(newType: String): Subject[IdState.Unminted] = {
+      // This should never be called with an unidentifiable subject, because the
+      // point of this is to fix references that have been identified, but have the wrong type,
+      // so let this error out if called on a subject with an inappropriate state
+      val identifiable =
+        original.asInstanceOf[Subject[IdState.Identifiable]]
+      val newSourceIdentifier =
+        mainSourceIdentifier.get.copy(ontologyType = newType)
+      identifiable
+        .copy(
+          id = identifiable.id
+            .copy(
+              sourceIdentifier = newSourceIdentifier
+            ),
+          concepts =
+            if (identifiable.concepts.length == 1) {
+              val oldConcept = identifiable.concepts.head
+              List(
+                AbstractRootConcept(
+                  newType,
+                  oldConcept.id.copy(sourceIdentifier = newSourceIdentifier),
+                  oldConcept.label
+                )
+              )
+            } else
+              identifiable.concepts
+        )
+    }
+  }
+
+  case class ContributorWithOntologyType(
+    original: Contributor[IdState.Unminted]
+  ) extends EntityWithOntologyType[Contributor[IdState.Unminted]] {
+
+    /**
+      * Return the contributor's Agent's sourceIdentifier.
+      * Contributors do not have a sourceIdentifier of their own, so the identifier is that of
+      * whatever resides within its agent property
+      * (Contributor _does_ extend HasId, but is always unidentifiable)
+      */
+    def mainSourceIdentifier: Option[SourceIdentifier] =
+      original.agent.id.allSourceIdentifiers.headOption
+
+    /**
+      * Return a copy of this Contributor with the Agent having been assigned the correct type.
+      */
+    def copyWithNewType(newType: String): Contributor[IdState.Unminted] = {
+      // This should never be called with an unidentifiable agent, because the
+      // point of this is to fix references that have been identified, but have the wrong type,
+      // so let this error out if called on a subject with an inappropriate state
+      val oldAgent =
+        original.agent
+          .asInstanceOf[AbstractRootConcept[IdState.Identifiable]]
+      val newSourceIdentifier =
+        mainSourceIdentifier.get.copy(ontologyType = newType)
```

Comment:
How do we know this will always be a `Some`? Can we reflect that knowledge in the types?